alpine/BumponTailInstabilityManager.h

```diff
@@ -0,0 +1,579 @@
+            epsilon = 0.1;
+            muBulk  = 0.0;
+            muBeam  = 4.0;
+            delta   = 0.01;
+        } else {
+            // Default value is two stream instability
+            kw      = 0.5;
+            sigma   = 0.1;
+            epsilon = 0.5;
+            muBulk  = -pi / 2.0;
+            muBeam  = pi / 2.0;
+            delta   = 0.01;
+        }
+	rmin(0.0);
+        rmax = 2 * pi / kw;
+
+        hr = rmax / nr;
+        // Q = -\int\int f dx dv
+        Q = std::reduce(rmax.begin(), rmax.end(), -1., std::multiplies<double>());
+        origin = rmin;
+        dt     = std::min(.05, 0.5 * *std::min_element(hr.begin(), hr.end()));
+        it     = 0;
+        time_m = 0.0;
+
+        m << "Discretization:" << endl
+          << "nt " << nt << " Np= " << totalP << " grid = " << nr << endl;
+
+        std::shared_ptr<Mesh_t<Dim>> mesh = std::make_shared<Mesh_t<Dim>>(domain, hr, origin);
+
+        isAllPeriodic = true;
+
+        std::shared_ptr<FieldLayout_t<Dim>> FL = std::make_shared<FieldLayout_t<Dim>>(MPI_COMM_WORLD, domain, decomp, isAllPeriodic);
+
+        std::shared_ptr<PLayout_t<T, Dim>> PL = std::make_shared<PLayout_t<T, Dim>>(*FL, *mesh);
+
+        if (solver == "OPEN") {
+            throw IpplException("BumpOnTailInstability", "Open boundaries solver incompatible with this simulation!");
+        }
+
+        setParticleContainer( std::make_shared<ParticleContainer_t>(PL) );
+
+        setFieldContainer( std::make_shared<FieldContainer_t>(hr, rmin, rmax, decomp) );
+
+        fcontainer_m->initializeFields(mesh, FL);
+
+        setFieldSolver( std::make_shared<FieldSolver_t>(this->solver, &fcontainer_m->getRho(), &fcontainer_m->getE()) );
+
+        fsolver_m->initSolver();
+
+        setLoadBalancer( std::make_shared<LoadBalancer_t>( lbt, fcontainer_m, pcontainer_m, fsolver_m) );
+
+        initializeParticles(mesh, FL);
+
+        if constexpr (EnablePhaseDump) {
+            if (ippl::Comm->size() != 1) {
+                m << "Phase dump only supported on one rank" << endl;
+                ippl::Comm->abort();
+            }
+            phase->initialize(*std::max_element(nr.begin(), nr.end()),
+                             *std::max_element(rmax.begin(), rmax.end()));
+        }
+
+        fcontainer_m->getRho() = 0.0;
+
+        fsolver_m->runSolver();
+
+        par2grid();
+
+        fsolver_m->runSolver();
+
+        grid2par();
+
+        dump();
+
+        m << "Done";
+    }
+
+    void initializeParticles(std::shared_ptr<Mesh_t<Dim>> mesh_m, std::shared_ptr<FieldLayout_t<Dim>> FL_m){
+        Inform m("Initialize Particles");
+
+        using DistR_t = ippl::random::Distribution<double, Dim, 2 * Dim, CustomDistributionFunctions>;
+        double* parR  = new double[2 * Dim];
+        parR[0]       = delta;
+        parR[1]       = kw[0];
+        parR[2]       = delta;
+        parR[3]       = kw[1];
+        parR[4]       = delta;
+        parR[5]       = kw[2];
```

Comment:
I enabled CG and tested the mini-apps. Also, I tested 1D and 2D cases of Bump on Tail and Landau Damping. It seems to reproduce the results for serial and multiple GPUs.

Reply:
Ok. What about Sonali's open boundary solver and P3M solver? If I am not wrong they were only for Penning trap I think. Could you please test that too? Also it would be good to test some of these cases on multiple CPU only builds too.

Reply:
It seems to reproduce the results of PenningTrap for P3M and OPEN as well (tested on cpu, openmp with 4 threads and 4 gpus compared to master's results)